src/main/java/teammates/ui/controller/AdminSearchPageAction.java

```diff
@@ -259,27 +263,29 @@ private String findAvailableInstructorGoogleIdForCourse(String courseId){
             return tempCourseIdToInstructorGoogleIdMap.get(courseId);
         }
         
-        String googleId = "";
-        
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
         
         if(instructorList == null || instructorList.isEmpty()){
-            return googleId;
+            return "";
         }
-        
-        for(InstructorAttributes instructor : instructorList){
-          
-            if(instructor.googleId != null){
-               googleId = instructor.googleId;
-               if(instructor.isAllowedForPrivilege(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)){            
-                   break;
-               } 
-            }            
+
+        for (InstructorAttributes instructor : instructorList) {
+
+            if (instructor.isRegistered() && instructor.hasCoownerPrivileges()) {
+                tempCourseIdToInstructorGoogleIdMap.put(courseId, instructor.googleId);
+                return instructor.googleId;
+            }
         }
-        
-        tempCourseIdToInstructorGoogleIdMap.put(courseId, googleId);
-        
-        return googleId; 
+
+        for (InstructorAttributes instructor : instructorList) {
+
+            if (instructor.isRegistered() && instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)) {
```

Comment:
Can you add an assertion in <code>isAllowedForPrivilege</code> that the input is a name of a privilege